### PR TITLE
fix: Delay container shutdown to allow fluent-bit to dequeue

### PR DESCRIPTION
**Pull request by JonZeolla**
# Contributor Comments

This includes a workaround to ensure that logs are dequeued prior to container shutdown, and adds a TRACE feature for more detailed debugging in the future.

## Pull Request Checklist

Thank you for submitting a contribution to `easy_infra`!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:

- [x] Rebase your branch against the latest commit of the target branch
- [x] If you are adding a dependency, please explain how it was chosen
- [x] If manual testing is needed in order to validate the changes, provide a testing plan and the expected results
- [x] If there is an issue associated with your Pull Request, link the issue to the PR.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -135,7 +137,6 @@ FROM minimal AS az
 USER root
 ARG AZURE_CLI_VERSION
 ENV AZURE_CLI_VERSION="${AZURE_CLI_VERSION}"
-SHELL ["/bin/bash", "-o", "pipefail", "-c"]
```
Comment (JonZeolla) [inline Dockerfile:138]: Duplicitive; already set in `minimal`

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -10,6 +10,11 @@ shopt -s dotglob
 # to stdout
 fluent-bit -c /usr/local/etc/fluent-bit/fluent-bit.conf --verbose 2>/dev/null
 
+if [[ "${LOG_LEVEL:-}" == "DEBUG" ]]; then
+  strace -o /tmp/strace-fluent-bit -e signal=all -p "$(pidof fluent-bit)" &
```
Comment (JonZeolla) [inline docker-entrypoint.sh:14]: Not showing the expected results, need to test on linux

File: fluent-bit.conf [hunk 3]
```diff
@@ -1,5 +1,6 @@
 [SERVICE]
     flush           .2
+    grace           2
```
Comment (JonZeolla) [inline fluent-bit.conf:3]: Doesn't seem to be helping

File: Dockerfile [hunk 4]
```diff
@@ -48,9 +48,11 @@ RUN groupadd --gid 53150 -r easy_infra \
                                                groff \
                                                jq \
                                                less \
+                                               libcap2-bin \
```
Comment (JonZeolla) [inline Dockerfile:51]: Perhaps we split these out to a `TRACE=true` build?

Comment (JonZeolla) [reply Dockerfile]: I did this

**Review by digeseiso — DISMISSED** [review]

**Review by digeseiso — APPROVED** [review]

**Pull request merged**